cpp/include/copc-lib/io/laz_writer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef COPCLIB_IO_LAZ_WRITER_H_
+#define COPCLIB_IO_LAZ_WRITER_H_
+
+#include <array>
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <stdexcept>
+#include <string>
+
+#include "copc-lib/copc/copc_config.hpp"
+#include "copc-lib/geometry/vector3.hpp"
+#include "copc-lib/io/laz_base_writer.hpp"
+#include "copc-lib/las/header.hpp"
+#include "copc-lib/las/laz_config.hpp"
+#include "copc-lib/las/points.hpp"
+#include "copc-lib/las/utils.hpp"
+
+namespace copc::laz
+{
+
+class LazWriter : public BaseWriter
+{
+
+  public:
+    LazWriter(std::ostream &out_stream, const las::LazConfigWriter &las_config);
+
+    // Write a group of points as a chunk
+    void WritePoints(const las::Points &points);
```

Comment:
I think we should also have a function that allows adding single points. We'd have to manage the chunks more manually though, so it might be too much complexity

Reply:
What do you think?

Reply:
We could simply keep a `Points` object as a buffer. When the user calls `WritePoint`, the point gets added to the writer's `Points` buffer. Then, the chunk gets written and the buffer cleared when either 1) the user calls `WritePoints` or 2) a certain number of points is reached

Reply:
Yeah that might be something we add to backlog, I don't really have a use for this right now. But would be good to have for general use for sure!

Reply:
k